sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -277,44 +273,29 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
     assert(nonSkewPartitionIndices.nonEmpty)
     val shouldCoalesce = conf.getConf(SQLConf.REDUCE_POST_SHUFFLE_PARTITIONS_ENABLED)
     if (!shouldCoalesce || nonSkewPartitionIndices.length == 1) {
-      Seq(SinglePartitionSpec(nonSkewPartitionIndices.head))
```

Comment:
This was a bug too? when `shouldCoalesce == false && nonSkewPartitionIndices.length > 1`, it should be `nonSkewPartitionIndices.map(SinglePartitionSpec)`, right?

Reply:
ah yes! it was introduced when we add the config...